pkg/autoscaler/stats_scraper.go

```diff
@@ -48,10 +49,20 @@ const (
 	scraperMaxRetries = 10
 )
 
+var (
+	// errFailedGetEndpoints specifies the error returned by scraper when it fails to
+	// get endpoints.
+	errFailedGetEndpoints = errors.New("failed to get endpoints")
```

Comment:
If you make these public, then the methods below will not be required
and you can just check the errors for equality since you are returning _the exact_ object.

This is usually how go does this kind of stuff  (e.g. https://golang.org/pkg/io/#pkg-variables)

Reply:
Cool!! That's good to know